Modules/FireForget/Get-AutorunsFF.ps1

```diff
@@ -0,0 +1,1763 @@
+                $Result = New-Object psobject
+                $Result | Add-Member -MemberType NoteProperty -Name 'Path' -Value $Resolved.__PATH
+                $Result | Add-Member -MemberType NoteProperty -Name 'Item' -Value $Resolved._Name
+                $Result | Add-Member -MemberType NoteProperty -Name 'Value' -Value $CLSID
+                $Result | Add-Member -MemberType NoteProperty -Name 'Category' -Value $Category
+                $Results += $Result
+            }
+        }
+    }
+
+    #Get File Path
+    foreach ($Entry in $Results) {
+        if ($Entry) {
+
+            $tmpVal = $Entry.Value -replace '"',""
+            if ($tmpVal) {
+                $tmpVal = $tmpVal
+                
+            } else {
+                $tmpVal = $null
+            }
+            $Entry | Add-Member -MemberType NoteProperty -Name 'FilePath' -Value $tmpVal
+        }
+    }
+    if ($Results) {Process-Results -Results $Results}
+}
+
+#Main Function
+$functionlist = @(
+    "Get-ASEPAppInitDLLs -ShowSignature -ShowFileHash",                    #0 67ms / 114ms  +Sig  +Hash
+    "Get-ASEPBootExecute -ShowSignature",                                  #1 194ms / 600ms  +Sig
+    "Get-ASEPExplorerAddons -ShowSignature -ShowFileHash -ShowEntropy",    #2 51s902ms / 3m48s668ms #no-go
+    "Get-ASEPSidebarGadgets",                                              #3 106ms / 83ms
+    "Get-ASEPImageHijacks",                                                #4 798ms / 779ms 
+    "Get-ASEPIEAddons -ShowSignature -ShowFileHash -ShowEntropy",          #5 506ms / 7s68ms   #no-go
+    "Get-ASEPKnownDLLs -ShowSignature",                                    #6 363ms / 12s707ms  +Sig  ***
+    "Get-ASEPLogon -ShowSignature",                                        #7 1s755ms / 38s352ms  +Sig  ###
+    "Get-ASEPWinsock",                                                     #8 639ms / 4s368ms
+    "Get-ASEPCodecs -ShowSignature -ShowFileHash -ShowEntropy",            #9 4s638ms / 52s164ms   #no-go
+    "Get-ASEPOfficeAddins -ShowSignature -ShowFileHash -ShowEntropy",      #10 5s799ms / 13s265ms  #no-go
+    "Get-ASEPPrintMonitors",                                               #11 234ms / 206ms ###
+    "Get-ASEPLSAProviders",                                                #12 204ms / 1s563ms
+    "Get-ASEPService -ShowSignature -ShowFileHash",                        #13 8s957ms / 1m25s657ms  +Sig  +Hash? ###
+    "Get-ASEPWinLogon -ShowSignature -ShowFileHash",                       #14 380ms / 8s324ms  +Sig  +Hash?  ###
+    "Get-ASEPWmi"                                                          #15 3m55s817ms / no friggin way   #no-go
+)
+
+Safe comprehensive spread
+0, 1, 3, 4, 6, 7, 8, 11, 12, 13, 14  | foreach {
+    iex $functionlist[$_]
+}
+
+#boot Logon spread
+#1, 7, 14 | foreach { iex $functionlist[$_] }
+#
```

Comment:
Wow. Interesting approach. Have you found it provides comprehensive enough coverage to replace the use of Autorunsc.exe?

Reply:
No, autorunsc.exe is much more thorough/efficient and easier to read/parse the output.  Certain organization (like ours) are VERY averse to letting us touch the disk at all for threat hunting activities.  Since the autorunsc.exe tool might not be on the list of approved software for an enterprise endpoint this module offers the convenient alternative of being ALL native powershell and giving "similar" capability.  As you can see from the comments, powershell is not terribly efficient at performing these checks and when running this module at scale some of the sub-functions took too much time/resources to justify running ALL of them without taking a really long time.

This module was developed long before some of our safety mechanisms that allowed the throttling of Kansa Fire&Forget jobs to only use idle CPU time and staggered-execution, so we could turn on all of the functions now without any trouble.  And if autorunsc.exe is "approved" for use on the endpoints and there are no concerns about dropping a binary to the disk then that method would definitely be preferred.